packages/core/src/codecs/json2.ts

```diff
@@ -0,0 +1,79 @@
+// Adapted from https://github.com/hms-dbmi/vizarr/blob/5b0e3ea6fbb42d19d0e38e60e49bb73d1aca0693/src/utils.ts#L26
+import type { Chunk, ObjectType } from "../metadata.js";
+import {
+	get_strides,
+	json_decode_object,
+	json_encode_object,
+} from "../util.js";
+
+type EncoderConfig = {
+	skipkeys?: boolean;
+	ensure_ascii?: boolean;
+	check_circular?: boolean;
+	allow_nan?: boolean;
+	sort_keys?: boolean;
+	indent?: number;
+	separators?: [string, string];
+};
+type DecoderConfig = {
+	strict?: boolean;
+};
+
+type JsonCodecConfig =
+	& {
+		encoding?: string;
+	}
+	& EncoderConfig
+	& DecoderConfig;
+
+export class JsonCodec {
+	kind = "array_to_bytes";
+
+	_indent: JsonCodecConfig["indent"];
+
+	constructor(
+		public configuration: JsonCodecConfig,
+	) {
+		// Reference: https://github.com/zarr-developers/numcodecs/blob/0878717a3613d91a453fe3d3716aa9c67c023a8b/numcodecs/json.py#L36
+		const {
+			// encoding = 'utf-8',
+			// skipkeys = false,
+			// ensure_ascii = true,
+			// check_circular = true,
+			// allow_nan = true,
+			// sort_keys = true,
```

Comment:
I'm not sure I understand from the docs. Do you think we should try to mimic the defaults here? It seems like we only account for `indent`, so should we throw for other config?

Reply:
I'm happy to merge a codec that does the minimal behavior, but it would be nice to have some error message for stuff we don't know how to account for.

Reply:
Thanks for the feedback, just implemented or added errors for these cases